src/CodeGen_Hexagon.cpp

```diff
@@ -914,6 +940,47 @@ void CodeGen_Hexagon::visit(const Call *op) {
             internal_assert(op->type.lanes()*2 == op->args[0].type().lanes());
             value = slice_vector(codegen(op->args[0]), 0, op->type.lanes());
             return;
+        } else if (op->is_intrinsic(Call::shuffle_vector)) {
+            // TODO: This implementation should move to the function
+            // shuffle_vector that should become virtual in the
+            // base class CodeGen_LLVM. dsharletg is working on
+            // this refactoring.
+            if(op->type.lanes() * 2 == op->args[0].type().lanes()) {
+                const int64_t *first_idx; first_idx = as_const_int(op->args[1]);
+                if (!first_idx || *first_idx > 1) {
+                    CodeGen_Posix::visit(op);
+                    return;
+                }
+                bool even = (*first_idx == 0) ? true : false;
+                bool use_vpack = true;
+                for (int i = 1; i < (int)op->args.size() - 1; i++) {
+                    const int64_t *curr = as_const_int(op->args[i]);
+                    const int64_t *next = as_const_int(op->args[i+1]);
+                    if (!(curr && next && ((*next) - (*curr)) == 2)) {
+                        use_vpack = false;
+                    }
+                }
+                if (use_vpack) {
+                    int lanes = op->type.lanes();
+                    Value *dv = codegen(op->args[0]);
+                    Value *low = slice_vector(dv, 0, lanes);
+                    Value *high = slice_vector(dv, lanes, lanes);
+                    Intrinsic::ID intrin_id;
+                    switch(op->type.bits()) {
+                    case 8: intrin_id = even ? IPICK(llvm::Intrinsic::hexagon_V6_vpackeb) :
+                        IPICK(llvm::Intrinsic::hexagon_V6_vpackob);
+                        break;
+                    case 16: intrin_id = even ? IPICK(llvm::Intrinsic::hexagon_V6_vpackeh) :
+                        IPICK(llvm::Intrinsic::hexagon_V6_vpackoh);
+                        break;
+                    }
+                    string s = "Generating a vpack";
+                    debug(4) << (s +  (even ? "e\n" : "o\n"));
```

Comment:
I see, the problem is that you wanted to add it. This line should just be:

    debug(4) << "Generating a vpack" << (even ? "e" : "o") << "\n";